### ISSUE TITLE
`show_window_menu` in `drag_window` example not working on Wayland

### ISSUE THREAD

**Issue by ids1024** [rust-windowing/winit#3466]
On cosmic-comp and Gnome Shell, `WAYLAND_DEBUG=1 cargo run --example drag_window` shows the `show_window_menu` request being made, but the menu doesn't appear.

I'm not really sure what's wrong. The serial and position seems right, and it's not clear what's different from the window decorations, where it does work.

**Comment by kchibisov** [comment]
If you can repro on gnome let me know, but it wasn't the case for me. There's nothing you can do different other than requesting it and there's no requirements in protocol to do anything extra, other than asking for it with _some_ serial from input event, which we do.

So sounds like cosmic bug.

**Comment by ids1024** [comment]
There could be something wrong with how cosmic-comp is handing things, but I see the same issue in Gnome Shell (45.3, on Fedora 39). Right clicking on the window decorations works (presumably handled by sctk-adwaita), but right clicking on the body of the window doesn't work, even though a `show_window_menu` request is sent.

**Comment by kchibisov** [comment]
it's a compositor policy to allow it or not. seems like gnome instantly dismisses it now when you release the button.

If you bind to `Press` it'll work.

**Comment by ids1024** [comment]
Ah, I just tested that. I thought I had already tried it, but it seems to work if changed to trigger on `Pressed` rather than `Released`.

It doesn't dismiss on button release. Mutter's `xdg_toplevel_show_window_menu` implementation calls `meta_wayland_seat_get_grab_info` and returns if that is `NULL`. Presumably the has ended if button release has been received.

I wonder if the protocol could be clearer on how this is expected to be handled by clients and servers, but it may be worth changing in the exactly and adding to the method's documentation.
